python/tests/test_resolve.py

```diff
@@ -25,3 +29,39 @@ async def test_resolve_default_model_config_with_openai_parser():
         "temperature": 1,
         "top_p": 1,
     }
+
+
+@pytest.mark.asyncio
+async def test_resolve_after_run(set_temporary_env_vars):
+    """
+    After running a prompt, resolving the same prompt should return the same result as if it was never run assuming no parameters or dependencies are at play.
+
+    This test is specific to the openai model parser
+    """
+    with patch.object(
+        openai.chat.completions, "create", side_effect=mock_openai_chat_completion
+    ):
```

Comment:
I think this is OK for now, but please consider passing this down explicitly (either manually or via a DI framework). 

Zen of python:

> Explicit is better than implicit.

Reply:
can you give me an example of what that looks like? I don't see how it would make sense to pass down the client or your inject all the way from config.run() to the end api call